src/wasm-binary.h

```diff
@@ -623,19 +623,36 @@ class WasmBinaryWriter : public WasmVisitor<WasmBinaryWriter, void> {
     finishSection(start);
   }
 
+  std::map<Name, uint32_t> mappedImports; // name of the Import => index
   uint32_t getImportIndex(Name name) {
-    // TODO: optimize
-    for (size_t i = 0; i < wasm->imports.size(); i++) {
-      if (wasm->imports[i]->name == name) return i;
+    if (mappedImports.size()) {
+      assert(mappedImports.count(name));
```

Comment:
literally the same two lines are outside of the if, so how about removing these two and flipping the if condition?